### ISSUE TITLE
Use the count of video to store it offline and serve it whenever opened.

### ISSUE THREAD

**Issue by Parth-Vader** [Parth-Vader/YoutubeDataCollector#3]
This would save your bandwidth.

Suppose I watch a video 10 times in 3 days - this would be detected by a script, and the video would be downloaded and served whenever this url would be opened the next time.

**Comment by aniketgattani** [comment]
@Parth-Vader we can use here a service worker and cache these videos whenever necessary. I would like to contribute to this issue. Do u want it any other way ?

**Comment by Parth-Vader** [comment]
Not sure - this is more of a `moonshot idea` for me. So you can start with service worker if you want to.

**Comment by aniketgattani** [comment]
But this would take up a lot of cache memory and please tell me when do i need to store them and should there be some restriction ?

**Comment by Parth-Vader** [comment]
@aniketgattani, For now, you can just use temporary data with a youtube link - maybe in a .csv file for now, but in another issue #2 - firebase has been decided to be used.
Try to download and serve one video as for now - if that's successful we can continue further. Does this sound reasonable?

**Comment by aniketgattani** [comment]
yeah okay so for now I will add the youtube link to the cache and then try to download it and will push a PR for the same. Plz tell me when do u want to save the video means after what checks. Also tell me if I got something wrong
